src/agoradatatools/great_expectations/gx/expectations/genes_biodomains.json

```diff
@@ -56,51 +56,75 @@
       "meta": {}
     },
     {
-      "expectation_type": "expect_column_values_to_have_list_length_in_range",
-      "kwargs": {
-        "column": "gene_biodomains",
-        "list_length_range": [
-          1,
-          19
-        ]
-      },
-      "meta": {}
-    },
-    {
-      "expectation_type": "expect_column_values_to_have_list_members_of_type",
-      "kwargs": {
-        "column": "gene_biodomains",
-        "member_type": "dict"
-      },
-      "meta": {}
-    },
-    {
-      "expectation_type": "expect_column_values_to_have_list_of_dict_with_expected_values",
+      "expectation_type": "expect_column_values_to_match_json_schema",
       "kwargs": {
         "column": "gene_biodomains",
-        "list_dict_values": {
-          "key": "biodomain",
-          "values": [
-            "Apoptosis",
-            "Vasculature",
-            "Lipid Metabolism",
-            "Proteostasis",
-            "Immune Response",
-            "Autophagy",
-            "Mitochondrial Metabolism",
-            "Structural Stabilization",
-            "Synapse",
-            "Endolysosome",
-            "Metal Binding and Homeostasis",
-            "Oxidative Stress",
-            "Epigenetic",
-            "APP Metabolism",
-            "Cell Cycle",
-            "DNA Repair",
-            "RNA Spliceosome",
-            "Tau Homeostasis",
-            "Myelination"
-          ]
+        "json_schema": {
+          "$id": "https://github.com/Sage-Bionetworks/agora-data-tools/src/agoradatatools/great_expectations/gx/json_schemas/genes_biodomains/gene_biodomains_schema.json",
+          "$schema": "https://json-schema.org/draft/2019-09/schema",
+          "default": [],
+          "items": {
+            "default": {},
+            "properties": {
+              "biodomain": {
+                "default": "",
+                "enum": [
+                  "Apoptosis",
+                  "Vasculature",
+                  "Lipid Metabolism",
+                  "Proteostasis",
+                  "Immune Response",
+                  "Autophagy",
+                  "Mitochondrial Metabolism",
+                  "Structural Stabilization",
+                  "Synapse",
+                  "Endolysosome",
+                  "Metal Binding and Homeostasis",
+                  "Oxidative Stress",
+                  "Epigenetic",
+                  "APP Metabolism",
+                  "Cell Cycle",
+                  "DNA Repair",
+                  "RNA Spliceosome",
+                  "Tau Homeostasis",
+                  "Myelination"
+                ],
+                "type": "string"
+              },
+              "go_terms": {
+                "default": [],
+                "items": {
+                  "UniqueItems": true,
+                  "maxItems": 2,
```

Comment:
why maxItems 2?

Reply:
To be honest, I don't remember if this was set when I generated the initial schema, or if I did this myself. Is there a different `maxItems` value I should change it to?

Reply:
This field can have any number of `go_terms` in it, so it might make sense to just remove the `maxItems` expectation. Unless there's a way to say the number of items should be equal to the number that's in the `n_gene_biodomain_terms` field. That sounds kind of complicated though.

Reply:
The current theoretical max possible value is 1383, but the current actual max in the data is 59. I think setting it to 100 should be fine for now, and possibly for all of time. If it starts failing someday, we can adjust the max value then.

Reply:
I guess I am curious why the expectation didn't fail, given that the configured max is significantly lower than what is in the data.

Reply:
Well, that's concerning. I'll look into that!

Reply:
Ah, the `minItems` and `maxItems` should not be inside of `items`, but rather one level up. It fails with `maxItems` set to 2 in the correct position but passes with `maxItems` set to 100.